controller/cmd/proxy-injector/main.go

```diff
@@ -1,19 +1,147 @@
 package proxyinjector
 
 import (
+	"encoding/base64"
+	"flag"
+	"fmt"
+	"log"
+	"strings"
+	"time"
+
+	"github.com/golang/protobuf/ptypes"
+	"github.com/linkerd/linkerd2/controller/gen/config"
 	"github.com/linkerd/linkerd2/controller/k8s"
 	injector "github.com/linkerd/linkerd2/controller/proxy-injector"
 	"github.com/linkerd/linkerd2/controller/webhook"
+	"github.com/linkerd/linkerd2/pkg/flags"
 )
 
 // Main executes the proxy-injector subcommand
 func Main(args []string) {
+
+	cmd := flag.NewFlagSet("proxy-injector", flag.ExitOnError)
+	controllerNamespace := cmd.String("controller-namespace", "", "namespace in which Linkerd is installed")
+	cniEnabled := cmd.Bool("cni-enabled", false, "network ranges for which the Linkerd proxy does destination lookups by IP address")
+	clusterDomain := cmd.String("cluster-domain", "", "kubernetes cluster domain")
+	identityScheme := cmd.String("identity-scheme", "", "scheme used for the identity issuer secret format")
+	trustDomain := cmd.String("identity-trust-domain", "", "configures the name suffix used for identities")
+	encodedIdentityTrustAnchorPEM := cmd.String("identity-trust-anchors-pem", "", "base64 encoded trust anchors certificate")
+	identityIssuanceLifeTime := cmd.String("identity-issuance-lifetime", "", "the amount of time for which the Identity issuer should certify identity")
+	identityClockSkewAllowance := cmd.String("identity-clock-skew-allowance", "", "the amount of time to allow for clock skew within a Linkerd cluster")
+	omitWebHookSideEffects := cmd.Bool("omit-webhook-side-effects", false, "omit the sideEffects flag in the webhook manifests")
+	proxyImageName := cmd.String("proxy-image", "", "linkerd proxy container image name")
+	proxyImagePullPolicy := cmd.String("proxy-image-pull-policy", "", "linkerd proxy image's pull policy")
+	proxyInitImageName := cmd.String("proxy-init-image", "", "linkerd proxy init container image name")
+	proxyInitImagePullPolicy := cmd.String("proxy-init-image-pull-policy", "", "linkerd proxy init image's pull policy")
+	controlPort := cmd.Uint64("control-port", 0, "proxy port to use for control")
+	ignoreInboundPorts := cmd.String("ignore-inbound-ports", "", "ports and/or port ranges (inclusive) that should skip the proxy and send directly to the application")
+	ignoreOutboundPorts := cmd.String("ignore-outbound-ports", "", "outbound ports and/or port ranges (inclusive) that should skip the proxy")
+	inboundPort := cmd.Uint64("inbound-port", 0, "proxy port to use for inbound traffic")
+	adminPort := cmd.Uint64("admin-port", 0, "proxy port to serve metrics on")
+	outboundPort := cmd.Uint64("outbound-port", 0, "proxy port to use for outbound traffic")
+	proxyUID := cmd.Int64("proxy-uid", 0, "run the proxy under this user ID")
+	proxyLogLevel := cmd.String("proxy-log-level", "", "log level for the proxy")
+	disableExternalProfiles := cmd.Bool("disable-external-profiles", false, "disable service profiles for non-Kubernetes services")
```

Comment:
should invert this flag to make it affirmative? `enable-external-profiles`?

Reply:
Yeah, makes sense as we are using `enable-external-profiles` in the CLI too.